### PR TITLE
[InstCombine] Clean up bitwise folds without one-use check

### PR DESCRIPTION
**Pull request by dtcxzyw**
This patch removes some bitwise folds that fail to check the one-use constraint on the operands.
See also the comments https://github.com/llvm/llvm-project/pull/77231#issuecomment-1904090035.

### CODE REVIEW

File: llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp [hunk 1]
```diff
@@ -3565,22 +3565,6 @@ Instruction *InstCombinerImpl::visitOr(BinaryOperator &I) {
     if (match(Op1, m_Xor(m_Specific(B), m_Specific(A))))
       return BinaryOperator::CreateOr(Op1, C);
 
-  // ((A & B) ^ C) | B -> C | B
-  if (match(Op0, m_c_Xor(m_c_And(m_Value(A), m_Specific(Op1)), m_Value(C))))
-    return BinaryOperator::CreateOr(C, Op1);
-
-  // B | ((A & B) ^ C) -> B | C
-  if (match(Op1, m_c_Xor(m_c_And(m_Value(A), m_Specific(Op0)), m_Value(C))))
-    return BinaryOperator::CreateOr(Op0, C);
```
Comment (nikic) [inline llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp:3574]: Aren't these two okay for multi-use?

**Review by nikic — COMMENTED** [review]
I think you may be able to drop https://github.com/llvm/llvm-project/blob/616ca003421f3114c8637a42c3d1cae09e20ef6e/llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp#L3621-L3630 as redundant? It has a one-use check, so I think it is subsumed by your fold.

**Comment by dtcxzyw** [comment]
> I think you may be able to drop
> 
> https://github.com/llvm/llvm-project/blob/616ca003421f3114c8637a42c3d1cae09e20ef6e/llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp#L3621-L3630
> 
> as redundant? It has a one-use check, so I think it is subsumed by your fold.

Done

**Review by nikic — APPROVED** [review]
LGTM

**Pull request merged**